src/Deserializers/StatementListDeserializer.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Wikibase\DataModel\Deserializers;
+
+use Deserializers\Deserializer;
+use Deserializers\Exceptions\DeserializationException;
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * Package private
+ *
+ * @licence GNU GPL v2+
+ * @author Bene* < benestar.wikimedia@gmail.com >
+ */
+class StatementListDeserializer implements Deserializer {
+
+	/**
+	 * @var Deserializer
+	 */
+	private $statementDeserializer;
+
+	/**
+	 * @param Deserializer $statementDeserializer
+	 */
+	public function __construct( Deserializer $statementDeserializer ) {
+		$this->statementDeserializer = $statementDeserializer;
+	}
+
+	/**
+	 * @see Deserializer::deserialize
+	 *
+	 * @param mixed $serialization
+	 *
+	 * @return object
+	 * @throws DeserializationException
+	 */
+	public function deserialize( $serialization ) {
+		$this->assertHasGoodFormat( $serialization );
+
+		return $this->getDeserialized( $serialization );
+	}
+
+	private function getDeserialized( array $serialization ) {
+		$statementList = new StatementList();
+
+		foreach( $serialization as $statementArray ) {
```

Comment:
I spend some time wondering why we have a list of lists here. Having some kind of hint that these are grouped by property id would be nice. Which also makes me wonder if we should just accept statements with a property id different than that of the list they are in. Though that is not something to change here if this just re-implements the current code.